reproducibility_project/src/analysis/sampler.py

```diff
@@ -0,0 +1,59 @@
+"""Use the pymbar package to perform decorrelated equilibration sampling."""
+
+import numpy as np
+from pymbar import timeseries
+
+from reproducibility_project.src.analysis.equlibration import is_equilibrated
+
+
+def sample_job(job, variable="potential_energy", threshold=0.75):
+    """Use the timeseries module from pymbar to perform statistical sampling.
+
+    The start, end and decorrleated step size of the production region are
+    added to the job document.
+
+    Parameters
+    ----------
+    job : signac.contrib.job.Job
+        The Job object.
+    variable : str; default "potential_energy"
+        The variable to be used in sampling.
+    threshold : float, optional, default=0.75
+        Fraction of data expected to be equilibrated.
+    """
+    try:
+        job.doc["sample_results"]
```

Comment:
Lets name it `sampling_results` to be a bit more descriptive, but other  than that, it looks great!